frontend/src/Interface/nmpInterface.tsx

```diff
@@ -0,0 +1,150 @@
+              distanceBtwnPlantsRows: string;
+              willPlantsBePruned: boolean;
+              whereWillPruningsGo: string;
+              willSawdustBeApplied: boolean;
+            },
+          ];
+          FeedForageAnalyses: [];
+          SoilTest?: boolean;
+          LeafTest: {
+            leafTissueP: number;
+            leafTissueK: number;
+          };
+          HasSoilTest?: boolean;
+          PreviousYearManureApplicationFrequency?: boolean;
+          PreviousYearManureApplicationNitrogenCredit?: boolean;
+          SoilTestNitrateOverrideNitrogenCredit?: boolean;
+          IsSeasonalFeedingArea?: boolean;
+          SeasonalFeedingArea?: boolean;
+          FeedingDaysSpentInFeedingArea?: boolean;
+          FeedingPercentageOutsideFeeingArea?: boolean;
+          MatureAnimalCount?: boolean;
+          GrowingAnimalCount?: boolean;
+          MatureAnimalAverageWeight?: boolean;
+          GrowingAnimalAverageWeight?: boolean;
+          MatureAnimalDailyFeedRequirementId?: number;
+          GrowingAnimalDailyFeedRequirementId: number;
+        },
+      ];
+      FarmAnimals: [];
+      FarmManures: [
+        {
+          Id: number;
+          Customized: boolean;
+          SourceOfMaterialId: string;
+          SourceOfMaterialStoredSystemId?: boolean;
+          SourceOfMaterialImportedManureId: number;
+          SourceOfMaterialName: string;
+          ManureId: number;
+          Name?: boolean;
+          ManureClass?: boolean;
+          SolidLiquid?: boolean;
+          Moisture?: boolean;
+          Nitrogen: number;
+          Ammonia: number;
+          Phosphorous: number;
+          Potassium: number;
+          DMId: number;
+          NMinerizationId: number;
+          Nitrate?: boolean;
+          Stored_Imported: number;
+          IsAssignedToStorage: boolean;
+          IncludedSourceOfMaterialIds: [];
+          GroupedWithCollectedAnalysisSourceItemIds: [];
+        },
+      ];
+      GeneratedManures: [];
+      ImportedManures: [
+        {
+          MaterialName: string;
+          ManureTypeName: string;
+          AnnualAmount: number;
+          AnnualAmountUSGallonsVolume: number;
+          AnnualAmountCubicYardsVolume: number;
+          AnnualAmountCubicMetersVolume: number;
+          AnnualAmountTonsWeight: number;
+          AnnualAmountDisplayVolume: string;
+          AnnualAmountDisplayWeight: string;
+          Units: number;
+          Moisture: number;
+          StandardSolidMoisture: number;
+          IsMaterialStored: boolean;
+          ManureId: string;
+          ManagedManureName: string;
+          Id: number;
+          ManureType: number;
+          AssignedToStoredSystem: boolean;
+          AssignedWithNutrientAnalysis: boolean;
+        },
+      ];
+      SeparatedSolidManures: [];
+      ManureStorageSystems: [];
+    },
+  ];
+  LastAppliedFarmManureId: null;
+  NMPReleaseVersion: number;
+}
+
+export default nmpInterface;
```

Comment:
the plugin was really helpful building this 🚀

Reply:
Actually did this with macros hehe :smile:

Reply:
Haha I was thinking of plugins of vscode